### PR TITLE
Fix RVA calculation

### PR DESCRIPTION
**Pull request by hillu**
This is my take on fixing #373, #399 while not breaking `pe.entry_point` calculation for tinype samples. This is still pretty much based on guesswork on my side. Suggestions on how to test my assumption are welcome!

### CODE REVIEW

File: libyara/modules/pe.c [hunk 1]
```diff
@@ -396,12 +402,20 @@ int64_t pe_rva_to_offset(
...
+        //
+        // Rounding everything less than 0x200 to 0 as discussed in
         // https://code.google.com/archive/p/corkami/wikis/PE.wiki#PointerToRawData
-        if (section_offset < 0x200)
-          section_offset = 0;
+        // does not work for PE32_FILE from the test suite and for
+        // some tinype samples where File Alignment = 4
+        // (http://www.phreedom.org/research/tinype/).
+        int alignment = OptionalHeader(pe, FileAlignment);
+        if (alignment) {
```
Comment (plusvic) [inline libyara/modules/pe.c]: Just a minor style-related change. Opening curling braces for ``if``, ``while``, ``switch``, etc, are put in the next line in all the codebase, please follow that style.

File: libyara/modules/pe.c [hunk 2]
```diff
@@ -1349,48 +1358,48 @@ void pe_parse_header(
...
       pe->object, "image_base");
 
   set_integer(
-      OptionalHeader(MajorLinkerVersion),
+      OptionalHeader(pe,MajorLinkerVersion),
       pe->object, "linker_version.major");
 
   set_integer(
-      OptionalHeader(MinorLinkerVersion),
+      OptionalHeader(pe,MinorLinkerVersion),
```
Comment (plusvic) [inline libyara/modules/pe.c]: Another minor style-related change: add a space after the comma.

**Comment by hillu** [comment]
@plusvic can you recommend a tool / configuration for automatically checking and/or reformatting source code?

**Comment by hillu** [comment]
@plusvic I think I fixed all the style issue.

I also did a bit of experimentation: On a Windows XP VM, I loaded
a `cmd.exe` (MD5:6d778e0f95447e6546553eeea709d03c) into OllyDbg.
The observed entry point address is 0xd4ad05046.

Static information obtained using `peinfo.py`:
FileAlignment: 0x200
AddressOfEntryPoint: 0x5046
ImageBase: 0x4AD00000
PointerToRawData for `.text` section: 0x400

The following rule matches for this file:

    rule xp_cmd_exe {
    strings: $ep = { 6a 28 68 68 51 d0 4a }
    condition: $ep at pe.entry_point
    }

I then edited PointerToRawData for the `.text` section to various
values and tried loading the resulting file into OllyDbg again.
Everything I tried between the original 0x400 and 0x5ff caused
OllyDbg to show the same entry point and the patched `cmd.exe`
would still run properly when started without a
debugger. (Anything < 0x400 or >= 0x600 would produce a
non-working binary.)

YARA 3.4.0 does not recognize the patched `cmd.exe` using the
above rule. Current master with my patches from this PR applied
does.

**Pull request merged**